tests/examples_methods_syntax/scatter_point_paths_hover.py

```diff
@@ -0,0 +1,149 @@
+"""
+Scatter plot with point paths on hover with search box
+======================================================
+This example demonstrates an interactive visualization technique inspired by [this Vega example](https://vega.github.io/vega/examples/global-development/), 
+adapted for Altair's higher-level API. Key features:
+
+1. Point Paths. On hover, shows data trajectories using a trail mark that thickens from past to present, clearly indicating the direction of time.
+2. Search Box. Implements a case-insensitive regex filter for country names, enabling dynamic, flexible data point selection to enhance exploratory analysis.
+
+"""
+# category: interactive charts
+
+import altair as alt
+from vega_datasets import data
+
+# Data source
+source = data.gapminder.url
+
+# X-value slider
+x_slider = alt.binding_range(min=1955, max=2005, step=5, name='Year ')
+x_select = alt.selection_point(name="x_select", fields=['year'], bind=x_slider, value=1980)
+
+# Hover selection
+hover = alt.selection_point(on='mouseover', fields=['country'], empty=False)
+# A separate hover for the points since these need empty=True  
+hover_point_opacity = alt.selection_point(on='mouseover', fields=['country'])  
+
+# Search box for country name
+search_box = alt.param(
+    value='',
+    bind=alt.binding(input='search', placeholder="Country", name='Search ')
+)
+
+# Base chart
+base = alt.Chart(source).encode(
+    x=alt.X('fertility:Q', scale=alt.Scale(zero=False, nice=False), title='Fertility'),
+    y=alt.Y('life_expect:Q', scale=alt.Scale(zero=False, nice=False), title='Life Expectancy'),
```

Comment:
I think the default behavior looks better than `nice=False` here. It make the dots less crammed in the top left corner:

nice=False
![image](https://github.com/vega/altair/assets/4560057/baa0ff2f-970b-438c-b2be-8a9494c9c14d)


Default:
![image](https://github.com/vega/altair/assets/4560057/5db9a28a-3827-40da-a0fd-10dd18f83daa)